src/utils/fs.ts

```diff
@@ -86,6 +88,16 @@ const resolveFunctionsDirectories = (input: string | string[]) => {
   return absoluteDirectories
 }
 
+const mkdirAndWriteFile: typeof pWriteFile = async (path, ...params) => {
+  if (typeof path === 'string') {
```

Comment:
Checking that path is a string because it can also be a file descriptor.

Reply:
If it is a file descriptor, shouldn't `pWriteFile()` be used instead of `mkdirAndWriteFile()` then?

Reply:
Absolutely, but we still need to account for the possibility that someone decides to use `mkdirAndWriteFile` with a file descriptor. If nothing else, TypeScript forces us to have this check since we're setting `mkdirAndWriteFile` to have the same type as `pWriteFile`.

Reply:
:+1: